src/commands/kv/bucket/mod.rs

```diff
@@ -32,43 +33,53 @@ pub fn directory_keys_values(
     directory: &Path,
     verbose: bool,
 ) -> Result<(Vec<KeyValuePair>, AssetManifest), failure::Error> {
-    let mut upload_vec: Vec<KeyValuePair> = Vec::new();
-    let mut asset_manifest: AssetManifest = AssetManifest::new();
+    match &fs::metadata(directory) {
+        Ok(file_type) if file_type.is_dir() => {
+            let mut upload_vec: Vec<KeyValuePair> = Vec::new();
+            let mut asset_manifest: AssetManifest = AssetManifest::new();
 
-    let dir_walker = get_dir_iterator(target, directory)?;
+            let dir_walker = get_dir_iterator(target, directory)?;
 
-    for entry in dir_walker {
-        let entry = entry.unwrap();
-        let path = entry.path();
-        if path.is_file() {
-            if verbose {
-                message::working(&format!("Preparing {}", path.display()));
-            }
+            for entry in dir_walker {
+                let entry = entry.unwrap();
+                let path = entry.path();
+                if path.is_file() {
+                    if verbose {
+                        message::working(&format!("Preparing {}", path.display()));
+                    }
 
-            validate_file_size(&path)?;
+                    validate_file_size(&path)?;
 
-            let value = std::fs::read(path)?;
+                    let value = std::fs::read(path)?;
 
-            // Need to base64 encode value
-            let b64_value = base64::encode(&value);
+                    // Need to base64 encode value
+                    let b64_value = base64::encode(&value);
 
-            let (url_safe_path, key) =
-                generate_path_and_key(path, directory, Some(b64_value.clone()))?;
+                    let (url_safe_path, key) =
+                        generate_path_and_key(path, directory, Some(b64_value.clone()))?;
 
-            validate_key_size(&key)?;
+                    validate_key_size(&key)?;
 
-            upload_vec.push(KeyValuePair {
-                key: key.clone(),
-                value: b64_value,
-                expiration: None,
-                expiration_ttl: None,
-                base64: Some(true),
-            });
+                    upload_vec.push(KeyValuePair {
+                        key: key.clone(),
+                        value: b64_value,
+                        expiration: None,
+                        expiration_ttl: None,
+                        base64: Some(true),
+                    });
 
-            asset_manifest.insert(url_safe_path, key);
+                    asset_manifest.insert(url_safe_path, key);
+                }
+            }
+            Ok((upload_vec, asset_manifest))
+        }
+        Ok(_file_type) => {
+            // any other file types (files, symlinks)
+            // TODO: return an error type here, like NotADirectoryError
+            Err(format_err!("site.bucket must be a directory"))
```

Comment:
Can we get a more actionable error message here that includes the value of what they have set as their bucket and direct them to their `wrangler.toml`? "site.bucket" is a bit opaque.